src/pages/settings/Profile/Contacts/NewContactMethodPage.js

```diff
@@ -76,13 +76,14 @@ function NewContactMethodPage(props) {
     }, [login, password, props.betas]);
 
     const submitForm = useCallback(() => {
+        const phoneLogin = LoginUtils.appendCountryCode(LoginUtils.getPhoneNumberWithoutSpecialChars(login));
+        const parsedPhoneNumber = parsePhoneNumber(phoneLogin);
+
         // If this login already exists, just go back.
         if (lodashGet(props.loginList, login)) {
```

Comment:
Instead of checking login, let's check userLogin where userLogin is either the plain login or the phone number in e164 format.

later on addNewContactMethodAndNavigate you can just use userLogin.